doc/README.md

```diff
@@ -1,13 +1,15 @@
-# Contributing to the Nixpkgs manual
+# Contributing to the Nixpkgs reference manual
 
-This directory houses the sources files for the Nixpkgs manual.
+This directory houses the sources files for the Nixpkgs reference manual.
 
-You can find the [rendered documentation for Nixpkgs `unstable` on nixos.org](https://nixos.org/manual/nixpkgs/unstable/).
-The rendering tool is [nixos-render-docs](../pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs), sometimes abbreviated `nrd`.
+Going forward, it should only contain [reference](https://nix.dev/contributing/documentation/diataxis#reference) documentation.
+For tutorials and guides, contribute to <https://nix.dev/> instead.
```

Comment:
I'm very dubious on the utility of having guides and reference documentation on totally separate domains, with separate styles and presentation. It undermines the utility of documentation to have it scattered around like this.

Is this reflective of an internal organizational boundary within Nix, [Conway's Law-style](https://en.wikipedia.org/wiki/Conway%27s_law)? I'm not sure why else references and guides would be separated like this.

Reply:
That's one aspect unfortunately, but some tutorial and guide material spans topics that don't necessarily belong in a single repo, topic-wise.
They also require a different kind of process than reference documentation.
The presentation style was inherited from nix.dev's origin as an independent project donated by Cachix.

Reply:
Either documentation should live under nixos.org or nix.dev, but the team should pick _one_ and stick to it. Having reference documentation and guides in separate repositories also makes documentation much more difficult to contribute to, as changes to one are often intertwined with changes to the other.

Reply:
> some tutorial and guide material spans topics that don't necessarily belong in a single repo, topic-wise.

Which material?

> They also require a different kind of process than reference documentation.

What are the process differences?

Reply:
> > don't necessarily belong in a single repo, topic-wise.

Most tutorials start from a low level of understanding, and provides info about Nix, as well as Nixpkgs and possibly other projects.
This might not apply to all tutorials, but managing this across many repos adds a lot of complexity for the docs team.

> > different kind of process than reference documentation.

Writing a good tutorial is much harder than writing reference docs. It requires more attention to the audience, their skill level, how it fits into the bigger picture of the set of tutorials.

See [The problem with tutorials](https://diataxis.fr/tutorials/#the-problem-of-tutorials) at diataxis.

Reply:
>  This might not apply to all tutorials, but managing this across many repos adds a lot of complexity for the docs team.

I agree — is the documentation team going to move the Nix manual, Nixpkgs manual, and NixOS manual into the nix.dev repo?

> Writing a good tutorial is much harder than writing reference docs.

I agree, but how is this related to what repository or domain the tutorials are maintained on?

Reply:
Reference docs should be maintained by contributors whenever they make changes.

Doing this for tutorials does not seems like something we should ask them to, because it's actually quite subtle and requires a more high level perspective that a contributor sure might have, but probably isn't on their mind that much while they work on their change.

Reply:
In the Nix team we actually do have practical and philosophical reasons to move the docs out of `nixos.org`.
After all Nix isn't just a part of NixOS. As for practicality we've had some issues that the nixos-homepage / marketing team aren't addressing. @fricklerhandwerk knows more.

Reply:
> Either documentation should live under nixos.org or nix.dev, but the team should pick _one_ and stick to it.

I'm all for that, I have a draft RFC for that here: https://github.com/nix-rfc-canonical-domain/rfcs/blob/canonical-domain/rfcs/1000-canonical-domain.md. Feel free to discuss in issues/PRs to that repo.

Reply:
> [Contributors maintaining] tutorials does not seems like something we should ask them to

That makes sense, but I'm not sure how you get from there to keeping reference material and tutorials in separate repositories/websites. It also makes it harder to link between the two sets of docs, as they're not guaranteed to be updated at the same time.

Reply:
> is the documentation team going to move the Nix manual, Nixpkgs manual, and NixOS manual into the nix.dev repo?

Yes, working on that. First step is the [Nix manual](https://nix.dev/reference/nix-manual) (currently as a copy), and the others need a bit more work to splice out of the nixos.org setup cleanly.

Reply:
As a contributor, keeping documentation in the nix.dev repo will make me far less likely to contribute to it, compared to the documentation being stored in the relevant repositories (Nix documentation in the Nix repository, Nixpkgs documentation in the Nixpkgs repository). Being able to write, review, and merge code and documentation changes together, in the same place, is a huge boon and not one I would lightly discard.

I'll note this is orthogonal to the documentation being on one domain; for example, right now, the Nix, NixOS, and Nixpkgs documentation lives on nixos.org, despite being built from several different repositories. It would be nice if nix.dev could continue that.

Reply:
To repeat in other words what @roberth wrote: Reference documentation should live with or in the code, no one disputes that. How to use the code with help of reference documentation in practice however is a different story altogether. 

We tend to have many typical use cases that mix and match Nix, Nixpkgs, and NixOS, and sometimes don't (actively) involve some of them at all. Historically this resulted in bits and pieces of documentation ending up in many different places, such as NixOS hacks and workarounds in the Nix manual, or very specific details about Nix in the NixOS manual. And, except for the NixOS Wiki, there was no place to drop integrated examples. This led many people to just write things on their personal blogs.

I see this is the primary reason for the current state of affairs, where users have to open a thousand tabs to find everything they need. In my opinion there are two ways out of that:
1. Only document things in a reference manual that actually concern that component, and have a place for overarching use cases.

    This is what we're doing right now. nix.dev is supposed to be the place for integration. It has the advantage that it caters better to both ends of the user spectrum: people who use everything in the Nix ecosystem indiscriminately, and in particular live on NixOS, and people who only use Nix as a tool embedded in some other ecosystem. Both get an overview in one place and can immediately dig down into specifics they care about.
    
    I think this strategy is slowly paying off.
 
2. Put everything in one big repository and put information where it seems fit, ideally in some structured format, and aggregate automatically.

   Asymptotically this would likely be a lot less effort for everyone. But in the short term it seems to be a huge infrastructure investment, and as long as that isn't made it would probably just worsen the structurelessness. We've in fact been trying to come up with cheap partial solutions in the past year, and then resorted to just writing more documentation instead. Not even speaking of making such a large change as moving Nix into the Nixpkgs monorepo being quite a challenge in terms of coordination.